### PR TITLE
Upgrade SD output image verification with CLIP score

### PR DESCRIPTION
**Pull request by MohitIntel**
Replaces existing inconsistent method of Stable Diffusion Model's output images verification by a more consistent quantitative CLIP score. 
Refer here for more details on implementation:
https://huggingface.co/docs/diffusers/conceptual/evaluation#text-guided-image-generation

### CODE REVIEW

File: tests/test_diffusers.py [hunk 1]
```diff
@@ -619,39 +621,34 @@ def test_no_generation_regression(self):
...
-                    0.67309505,
-                    0.70057267,
-                    0.6582885,
-                    0.6325019,
-                    0.6708976,
-                    0.6226433,
-                    0.58038336,
-                ]
-            )
+            target_score = 30.2968
```
Comment (MohitIntel) [inline tests/test_diffusers.py:634]: Bottomline is chosen as 95% of the mean of the CLIP scores I got from multiple runs (5-10 runs).

File: tests/test_diffusers.py [hunk 2]
```diff
@@ -619,39 +621,34 @@ def test_no_generation_regression(self):
...
         else:
-            expected_slice = np.array(
-                [0.70760196, 0.7136303, 0.7000798, 0.714934, 0.6776865, 0.6800843, 0.6923707, 0.6653969, 0.6408076]
-            )
+            target_score = 36.774
```
Comment (MohitIntel) [inline tests/test_diffusers.py:636]: Bottomline is chosen as 95% of the mean of the CLIP scores I got from multiple runs (5-10 runs).

File: tests/test_diffusers.py [hunk 3]
```diff
@@ -663,61 +660,29 @@ def test_no_generation_regression_ldm3d(self):
...
-                    0.6852779,
-                    0.6836072,
-                    0.6888298,
-                    0.6895473,
-                    0.6853674,
-                    0.67561126,
-                    0.660434,
-                ]
-            )
+            target_score = 29.63
```
Comment (MohitIntel) [inline tests/test_diffusers.py:673]: Bottomline is chosen as 95% of the mean of the CLIP scores I got from multiple runs (5-10 runs).

File: tests/test_diffusers.py [hunk 4]
```diff
@@ -663,61 +660,29 @@ def test_no_generation_regression_ldm3d(self):
...
-                    0.9184986,
-                    0.91994286,
-                    0.9242079,
-                    0.93387043,
-                    0.92345214,
-                    0.93558526,
-                    0.9223714,
-                ]
-            )
+            target_score = 35.81
```
Comment (MohitIntel) [inline tests/test_diffusers.py:675]: Bottomline is chosen as 95% of the mean of the CLIP scores I got from multiple runs (5-10 runs).

**Review by ssarkar2 — COMMENTED** [review]
@MohitIntel  can your please run style check

**Review by libinta — APPROVED** [review]

**Review by ssarkar2 — APPROVED** [review]

File: tests/test_diffusers.py [hunk 5]
```diff
@@ -619,39 +621,34 @@ def test_no_generation_regression(self):
...
-                [0.70760196, 0.7136303, 0.7000798, 0.714934, 0.6776865, 0.6800843, 0.6923707, 0.6653969, 0.6408076]
-            )
+            target_score = 36.774
+
         image = outputs.images[0]
         pil_image = numpy_to_pil(image)[0]
         pil_image.save("test_no_generation_regression_output.png")
 
+        clip_score = calculate_clip_score(np.expand_dims(image, axis=0), [prompt])
+        print(f"CLIP score: {clip_score}")
```
Comment (regisss) [inline tests/test_diffusers.py:643]: I would remove this print. We will get the value anyway if the assert fails.

File: tests/test_diffusers.py [hunk 6]
```diff
@@ -663,61 +660,29 @@ def test_no_generation_regression_ldm3d(self):
...
-                    0.9223714,
-                ]
-            )
+            target_score = 35.81
+
         rgb = outputs.rgb[0]
         depth = outputs.depth[0]
 
+        rgb_clip_score = calculate_clip_score(np.expand_dims(rgb, axis=0), [prompt])
+        print(f"RGB CLIP score: {rgb_clip_score}")
```
Comment (regisss) [inline tests/test_diffusers.py:681]: Same

**Review by regisss — COMMENTED** [review]
Why not, but that doesn't solve the original issue IMO. The reason these tests are inconsistent is that some SD pipelines don't generate reproducible outputs (mainly on Gaudi2). Are we okay with that?

File: tests/test_diffusers.py (hunk 1 above)
Comment (regisss) [reply tests/test_diffusers.py:634]:
```suggestion
            target_score = 29.8925
```

File: tests/test_diffusers.py (hunk 3 above)
Comment (regisss) [reply tests/test_diffusers.py:672]:
```suggestion
            target_score = 28.0894
```

**Review by regisss — COMMENTED** [review]
Let's rather put the min scores. I did a couple of suggestions for the two tests I saw failing.
I simply want to avoid having flaky tests.

**Review by regisss — APPROVED** [review]

**Pull request merged**